src/Security/Lti/OidcUserAuthenticator.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/**
+ *  This program is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU General Public License
+ *  as published by the Free Software Foundation; under version 2
+ *  of the License (non-upgradable).
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ *  Copyright (c) 2020 (original work) Open Assessment Technologies S.A.
+ */
+
+declare(strict_types=1);
+
+namespace OAT\SimpleRoster\Security\Lti;
+
+use OAT\Library\Lti1p3Core\Security\User\UserAuthenticationResult;
+use OAT\Library\Lti1p3Core\Security\User\UserAuthenticationResultInterface;
+use OAT\Library\Lti1p3Core\Security\User\UserAuthenticatorInterface;
+use OAT\Library\Lti1p3Core\User\UserIdentity;
+use OAT\SimpleRoster\DataTransferObject\LoginHintDto;
+use OAT\SimpleRoster\Entity\User;
+use OAT\SimpleRoster\Exception\LineItemNotFoundException;
+use OAT\SimpleRoster\Lti\Exception\InvalidGroupException;
+use OAT\SimpleRoster\Lti\Extractor\LoginHintExtractor;
+use OAT\SimpleRoster\Repository\UserRepository;
+
+class OidcUserAuthenticator implements UserAuthenticatorInterface
+{
+    public const UNDEFINED_USERNAME = 'Undefined Username';
+
+    /** @var LoginHintExtractor */
+    private $loginHintExtractor;
+
+    /** @var UserRepository */
+    private $userRepository;
+
+    public function __construct(LoginHintExtractor $loginHintExtractor, UserRepository $userRepository)
+    {
+        $this->loginHintExtractor = $loginHintExtractor;
+        $this->userRepository = $userRepository;
+    }
+
+    public function authenticate(string $loginHint): UserAuthenticationResultInterface
```

Comment:
You should wrap the logic within a try catch block to make sure you are not bubbling up any exceptions. In case of error the interface expects to send a `success = false` within the result without user identity.

Reply:
Also we need to add logs here about the result of the authentication

Reply:
actually you should let bubble up the exceptions, they will be catched and stamped 401 in the bundle automated endpoint. 

see https://github.com/oat-sa/bundle-lti1p3/blob/master/Action/Platform/Message/OidcAuthenticationAction.php#L48

good idea for the logger, always help

Reply:
Thx @ekkinox !